testsuite/tests/lazy/lazy3.ml

```diff
@@ -2,6 +2,17 @@
  ocamlopt_flags += " -O3 ";
 *)
 
+(* In this test we force a lazy from two concurrent domains without
+   synchronization. This leads to unspecified behavior but still
+   should not crash. Currently, the implementation raises Undefined,
+   and that's what we test here.
+
+   Note: due to a bug in the current implementation of
```

Comment:
This comment is likely to go out of sync. When we fix this bug, it is likely that we will not fix this comment as the existing tools (such as running the testsuite) won't help catch this. 

My preference should be to remove this Note altogether, with the idea that memory cleanup at exit feature will have a recommendation something to the effect of "When used in a multi-domain program, it is undefined behaviour not to join all the spawned domain before the main domain terminates. Undefined behaviour here includes crashes.". 

Additionally, It is not clear what this bug is. If we are keeping this comment in some form, it would be useful to link to the bug report somewhere (or the wip PR #12964).

Reply:
This makes sense, so I removed the `merge-me` label for now. (I am not patient enough to wait for @damiendoligez to update his PR, but then I am not motivated enough to fix the comment myself, so for now let's just wait.)

Reply:
I decided to go ahead and merge, and will remove the Note as recommended in trunk.